test/integration/controllermanager/project/activity/activity_test.go

```diff
@@ -95,6 +100,28 @@ var _ = Describe("Project Activity controller tests", Ordered, func() {
 			Expect(testClient.Update(ctx, obj)).To(Succeed())
 			log.Info("Updated object", "kind", kind, "object", client.ObjectKeyFromObject(obj))
 
+			By("Wait until manager has observed updated" + kind)
+			Eventually(func(g Gomega) {
+				switch currentObj := obj.(type) {
+				case *corev1.Secret:
+					updatedSecret := &corev1.Secret{}
+					g.Expect(mgrClient.Get(ctx, client.ObjectKeyFromObject(obj), updatedSecret)).To(Succeed())
+					g.Expect(currentObj.Data).To(DeepEqual(updatedSecret.Data))
+				case *gardencorev1beta1.BackupEntry:
+					updatedBackupEntry := &gardencorev1beta1.BackupEntry{}
+					g.Expect(mgrClient.Get(ctx, client.ObjectKeyFromObject(obj), updatedBackupEntry)).To(Succeed())
+					g.Expect(currentObj.Spec).To(DeepEqual(updatedBackupEntry.Spec))
+				case *gardencorev1beta1.Quota:
+					updatedQuota := &gardencorev1beta1.Quota{}
+					g.Expect(mgrClient.Get(ctx, client.ObjectKeyFromObject(obj), updatedQuota)).To(Succeed())
+					g.Expect(currentObj.Spec).To(DeepEqual(updatedQuota.Spec))
+				case *gardencorev1beta1.Shoot:
+					updatedShoot := &gardencorev1beta1.Shoot{}
+					g.Expect(mgrClient.Get(ctx, client.ObjectKeyFromObject(obj), updatedShoot)).To(Succeed())
+					g.Expect(currentObj.Spec).To(DeepEqual(updatedShoot.Spec))
+				}
```

Comment:
Can you simplify this switch? 
I suggest comparing the `resourceVersion` of the object in the manager cache with the `resourceVersion` of the version after the previous update call. That should be sufficient for this case.
Alternatively, checking deep equality of the entire object should also work.